src/app/students/(hooks)/useStudents.tsx

```diff
@@ -1,38 +1,50 @@
 'use client'
-import { useState, createContext, useContext, ChangeEvent, PropsWithChildren, useMemo } from 'react'
+import { useState, createContext, useContext, PropsWithChildren, useMemo } from 'react'
 import { Student } from '@/_temp_types/student'
 
-type DropdownOption  = {
-  label: string;
-  value: string;
+type DropdownOption = {
+    label: string;
+    value: string;
 }
 
 type StudentsContextType = {
-  displayStudents: Student[];
-  currentSections: DropdownOption[];
+    displayStudents: Student[];
+    currentSections: DropdownOption[];
+    fetchStudents: () => Promise<void>;
 }
-
 const StudentsContext = createContext<StudentsContextType | undefined>(undefined)
 
 const useStudentsProvider = (): StudentsContextType => {
-    const [studentsParse, setStudentsParse] = useState<Student[]>([])
     const [displayStudents, setDisplayStudents] = useState<Student[]>([])
     const [sections, setSections] = useState<DropdownOption[]>([])
+    const fetchStudents = async () => {
+        const fixedCourseNum = 1  // TODO: replace with actual course number
+        const baseURL = process.env.NEXT_PUBLIC_DB_HOST
+        const courseMemberData = await fetch(`${baseURL}/api/v1/course-members/course/${fixedCourseNum}/`).then(res => res.json())
```

Comment:
is this still the url structure?

Reply:
Yeah it's still the url structure for the frontend. I think the final touch should be to configure the endpoint so that it's the same on backend and frontend: `course/[id]/students/?queryparams`

Want to configure the endpoint on backend first though.